### DISCUSSION TITLE
Replicate pytest output for VS Code test results

### DISCUSSION THREAD

**Discussion by eleanorjboyd** [pytest-dev/pytest#11344] [Q&A]
Hello again, I am working on the rewrite for pytest testing in vscode and was hoping for a bit more advice. I appreciate all the help this community has given me so far and hope that these changes in VS Code are useful to you too! As outlined in this issue https://github.com/microsoft/vscode-python/issues/21861, I am redoing the look of the testing output for vscode. Before our options with the output were limited because we were parsing the output to populate our UI but the rewrite no longer does this due to a plugin we created. This being said we can now strive to have the best output possible, and for pytest this means most similar to the behavior users will expect from the command line. I was looking through your source code and was struggling to find how you craft the output to then print. I specifically want to produce the overall output per pytest run command that includes "platform, rootdir, config file etc" that is printed at the top of all pytest output. I would also like to be able to present per-test output to the user. This means for each test function run the out / errors / logs. I can see from the [TestReport item](https://docs.pytest.org/en/7.1.x/reference/reference.html#testreport) the existence of caplog:'', capstderr:'', and capstdout:'', which is likely what I will use but want to format it as similar as possible to how pytest would. Thanks!

Here is the output for the whole pytest run I was talking about
```
(.venv) eleanorboyd@Eleanors-MacBook-Pro test_directory % python -m pytest test_nope.py::test_this
=================================================== test session starts ====================================================
platform darwin -- Python 3.10.8, pytest-7.4.0, pluggy-1.2.0
rootdir: /Users/eleanorboyd/testingFiles/from_users/test_directory
plugins: describe-2.1.0
collected 1 item                                                                                                           

test_nope.py F                                                                                                       [100%]

========================================================= FAILURES =========================================================
xxxx per test info here xxx
--------------------------------------------------- Captured stdout call ---------------------------------------------------
this is print
================================================= short test summary info ==================================================
FAILED test_nope.py::test_this - assert 1 == 2
==================================================== 1 failed in 0.03s =====================================================
```

and a single test output
```

    def test_this():
        print("this is print")
        a = 1
>       assert a == 2
E       assert 1 == 2

test_nope.py:17: AssertionError
```

**Comment by nicoddemus** [comment]
Hi @eleanorjboyd,

All the terminal output code lies in the [terminal plugin](https://github.com/pytest-dev/pytest/blob/main/src/_pytest/terminal.py), so much that the official way to disable all pytest output is by passing `-p no:terminal` to pytest.
